osquery/tables/system/windows/drivers.cpp

```diff
@@ -48,46 +49,54 @@ const std::string kServiceKeyPath =
 
 static inline void win32LogWARNING(const std::string& msg,
                                    const DWORD code = GetLastError(),
-                                   const std::string& deviceName = "") {
-  LOG(WARNING) << msg << " for device " << deviceName
+                                   const std::string& device_name = "") {
+  LOG(WARNING) << msg << " for device " << device_name
                << ", error code: " + std::to_string(code);
 }
 
+// Unify the image path as systemRoot can contain systemroot and/or system32
+static inline std::string kNormalizeImage(std::string& path) {
+  std::transform(path.begin(), path.end(), path.begin(), ::tolower);
+  char sys_root[MAX_PATH] = {0};
+  GetSystemDirectory(sys_root, MAX_PATH);
+  return sys_root +
+         boost::regex_replace(path, boost::regex("^.*?system32"), "");
```

Comment:
In addition operation sys_root converts to `std::string` before. And the summary result is another string object. You could avoid this 2 additional allocations by using `std::string` for `sys_root`. If you like to :)
```c++
auto img_path = std::string(MAX_PATH, '\0');
auto size = GetSystemDirectory(&img_path.front(), img_path.size());
sys_root.resize(size);
sys_root.append(boost::regex_replace(path, boost::regex("^.*?system32"), ""));
return sys_root;
```